cpdb/cpdb.c

```diff
@@ -258,10 +258,12 @@ char *cpdbGetSysConfDir()
         free(env_xcd);
     }
         
-    config_dir = "/etc/cpdb";
+    const char* static_path = "/etc/cpdb";
+    config_dir = g_strdup(static_path);
     if (access(config_dir, R_OK) == 0 || mkdir(config_dir, CPDB_SYSCONFDIR_PERM) == 0)
```

Comment:
Could be slightly simplified by using `static_path` for the checks, and only duplicating the string in the `if` block : `return g_strdup(static_path);`.

(No need to free below then).

There's still a leak of `config_dir` further up and a part of the implementation of this function looks  "suspicious" to me, but that's unrelated to this change. I'll submit a separate PR.

Reply:
> Could

Done

Reply:
> Could be slightly simplified by using `static_path` for the checks, and only duplicating the string in the `if` block : `return g_strdup(static_path);`.
> 
> (No need to free below then).
> 
> There's still a leak of `config_dir` further up and a part of the implementation of this function looks "suspicious" to me, but that's unrelated to this change. I'll submit a separate PR.

Done